src/lava/lib/optimization/apps/scheduler/solver.py

```diff
@@ -0,0 +1,289 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+
+import networkx as ntx
+import numpy as np
+
+from networkx.algorithms.approximation import maximum_independent_set
+from typing import List, Dict, Tuple, Optional
+
+import matplotlib.pyplot as plt
+from matplotlib.patches import PathPatch
+from matplotlib.path import Path
+
+from lava.utils import loihi
+from lava.lib.optimization.apps.scheduler.problems import \
+    (SchedulingProblem, SatelliteScheduleProblem)
+from lava.lib.optimization.apps.scheduler.utils.util_funcs import *
+from lava.lib.optimization.problems.problems import QUBO
+from lava.lib.optimization.solvers.generic.solver import (OptimizationSolver,
+                                                          SolverConfig)
+from lava.lib.optimization.utils.generators.mis import MISProblem
+
+
+class Scheduler:
+
+    def __init__(self,
+                 sp: SchedulingProblem,
+                 qubo_weights: Tuple[int, int] = (1, 8),
+                 probe_cost: bool = False,
+                 probe_loihi_exec_time=False,
+                 probe_loihi_energy=False):
+        """Solver for Scheduling Problems.
+
+        Parameters
+        ----------
+        sp (SchedulingProblem) : Scheduling problem object as defined in
+        lava.lib.optimization.apps.scheduler.problems
+        qubo_weights (tuple(int, int)) : The QUBO weight matrix parameters
+        for diagonal and off-diagonal weights. Default is (1, 8).
+        probe_cost (bool) : Toggle whether to probe cost during the solver
+        run. Default is False.
+        """
+        self._problem = sp
+        self._graph = sp.graph
+        self._qubo_hyperparams = {
```

Comment:
I would suggest that we don't hardcode any QUBO parameters here, as the QUBO model will still change quite a bit. For example, the next generation of QUBO will not feature a refract hyperparameter anymore.
Thus, the Scheduling Solver should not implicitly assume any QUBO solver design

Reply:
We'll update the app solver as the underlying QUBO solver changes. From UX perspective, I want the solver to run even if a user provides no hyperparameters at the top-most level.

Reply:
Is there any unit test that catches this? Because I personally will forget to update this whenever I make changes to the QUBO solver.

Reply:
Yup. If the hyperparameters change, the solver unittest would fail.